packages/camera/camera_avfoundation/ios/Classes/CameraPlugin.m

```diff
@@ -148,274 +138,355 @@ - (void)availableCamerasWithCompletion:
-        }
-      };
-      _camera.dartAPI = [[FCPCameraEventApi alloc]
-          initWithBinaryMessenger:_messenger
-             messageChannelSuffix:[NSString stringWithFormat:@"%ld", cameraId]];
-      [_camera reportInitializationState];
-      [self sendDeviceOrientation:[UIDevice currentDevice].orientation];
-      [_camera start];
-      result(nil);
-    } else if ([@"takePicture" isEqualToString:call.method]) {
-      [_camera captureToFile:result];
-    } else if ([@"dispose" isEqualToString:call.method]) {
-      [_registry unregisterTexture:cameraId];
-      [_camera close];
-      result(nil);
-    } else if ([@"prepareForVideoRecording" isEqualToString:call.method]) {
-      [self.camera setUpCaptureSessionForAudio];
-      result(nil);
-    } else if ([@"startVideoRecording" isEqualToString:call.method]) {
-      BOOL enableStream = [call.arguments[@"enableStream"] boolValue];
-      if (enableStream) {
-        [_camera startVideoRecordingWithResult:result messengerForStreaming:_messenger];
+      if (error) {
+        completion(nil, error);
       } else {
-        [_camera startVideoRecordingWithResult:result];
-      }
-    } else if ([@"stopVideoRecording" isEqualToString:call.method]) {
-      [_camera stopVideoRecordingWithResult:result];
-    } else if ([@"pauseVideoRecording" isEqualToString:call.method]) {
-      [_camera pauseVideoRecordingWithResult:result];
-    } else if ([@"resumeVideoRecording" isEqualToString:call.method]) {
-      [_camera resumeVideoRecordingWithResult:result];
-    } else if ([@"getMaxZoomLevel" isEqualToString:call.method]) {
-      [_camera getMaxZoomLevelWithResult:result];
-    } else if ([@"getMinZoomLevel" isEqualToString:call.method]) {
-      [_camera getMinZoomLevelWithResult:result];
-    } else if ([@"setZoomLevel" isEqualToString:call.method]) {
-      CGFloat zoom = ((NSNumber *)argsMap[@"zoom"]).floatValue;
-      [_camera setZoomLevel:zoom Result:result];
-    } else if ([@"setFlashMode" isEqualToString:call.method]) {
-      [_camera setFlashModeWithResult:result mode:call.arguments[@"mode"]];
-    } else if ([@"setExposureMode" isEqualToString:call.method]) {
-      [_camera setExposureModeWithResult:result mode:call.arguments[@"mode"]];
-    } else if ([@"setExposurePoint" isEqualToString:call.method]) {
-      BOOL reset = ((NSNumber *)call.arguments[@"reset"]).boolValue;
-      double x = 0.5;
-      double y = 0.5;
-      if (!reset) {
-        x = ((NSNumber *)call.arguments[@"x"]).doubleValue;
-        y = ((NSNumber *)call.arguments[@"y"]).doubleValue;
-      }
-      [_camera setExposurePointWithResult:result x:x y:y];
-    } else if ([@"getMinExposureOffset" isEqualToString:call.method]) {
-      result(@(_camera.captureDevice.minExposureTargetBias));
-    } else if ([@"getMaxExposureOffset" isEqualToString:call.method]) {
-      result(@(_camera.captureDevice.maxExposureTargetBias));
-    } else if ([@"getExposureOffsetStepSize" isEqualToString:call.method]) {
```

Comment:
This method does not exist in the Pigeon version because hard-coding 0 as the return value is trivial to do in Dart.